### PR TITLE
Added 'multipart?' request helper

### LINKED ISSUE #1494: Add new multipart? request helper
We have a list of these helper methods that can be used in your Actions like `html?` or `ajax?`. While working on Breeze I came up with a use case for `multipart?` which should return true if the `"Content-Type"` header is `multipart/form-data;`. 

This could be used in before action pipes to do specific things if you're doing a multipart POST.

https://github.com/luckyframework/lucky/blob/master/src/lucky/request_type_helpers.cr

### PR DESCRIPTION
**Pull request by nodeMD**
## Purpose
Fixes #1494 

## Description
Adds 'multipart?' request helper

## Checklist
* [x] - An issue already exists detailing the issue/or feature request that this PR fixes
* [x] - All specs are formatted with `crystal tool format spec src`
* [x] - Inline documentation has been added and/or updated
* [x] - Lucky builds on docker with `./script/setup`
* [x] - All builds and specs pass on docker with `./script/test`

### CODE REVIEW

File: src/lucky/request_type_helpers.cr [hunk 1]
```diff
@@ -91,6 +91,13 @@ module Lucky::RequestTypeHelpers
     request.headers["X-Requested-With"]?.try(&.downcase) == "xmlhttprequest"
   end
 
+  # Check if the request is multipart
+  #
+  # This tests if the `Content-Type` header is `multipart/form-data`
+  def multipart? : Bool
+    request.headers["Content-Type"]?.try(&.downcase) == "multipart/form-data"
```
Comment (jwoertink) [inline src/lucky/request_type_helpers.cr:98]: I believe that most content-types won't be just `multipart/form-data`. Generally these will look more like

```
multipart/form-data; boundary=------boundary123
```

What I had to do in a recent commit was use `starts_with?` https://github.com/luckyframework/breeze/blob/master/src/breeze/actions/mixins/action_helpers.cr#L76-L78

**Review by jwoertink — CHANGES_REQUESTED** [review]
This looks like another great contribution! Just one small change to make sure we can catch a wider range.

File: src/lucky/request_type_helpers.cr (hunk 1 above)
Comment (nodeMD) [reply src/lucky/request_type_helpers.cr]: Thanks 👍 I read a little more about it and I see now how multipart is working 🎉

File: spec/lucky/request_type_helper_spec.cr [hunk 2]
```diff
@@ -71,7 +71,7 @@ describe Lucky::RequestTypeHelpers do
     action = FakeAction.new
     action.multipart?.should(be_false)
 
-    action.context.request.headers["Content-Type"] = "multipart/form-data"
+    action.context.request.headers["Content-Type"] = "multipart/form-data; boundary="
```
Comment (jwoertink) [inline spec/lucky/request_type_helper_spec.cr:74]: ah nice! 👍

**Review by jwoertink — APPROVED** [review]
Great work! Thanks for taking care of this.

**Pull request merged**